GameOfRunes/GameOfRunes/GameEngine/Systems/MoveSystem.swift

```diff
@@ -85,3 +86,46 @@ class MoveSystem: GKComponentSystem<MoveComponent>, System {
         super.removeComponent(component)
     }
 }
+
+/** Extension for Power Up implementations */
+extension MoveSystem {
+    private func checkFireVortexCollision() {
+        guard let hellfireComponent = gameEngine?.entities(for: .player)
+            .compactMap({ $0 as? HellfirePowerUpEntity }).first,
+            let hellfireNode = hellfireComponent.component(ofType: SpriteComponent.self)?.node else {
+                return
+        }
+        
+        for enemyEntity in gameEngine?.entities(for: .enemy) ?? [] {
+            guard enemyEntity.component(ofType: MoveComponent.self) != nil else {
+                continue
+            }
+            
+            if enemyEntity
+                .component(ofType: SpriteComponent.self)?
+                .node
+                .calculateAccumulatedFrame()
+                .intersects(hellfireNode.calculateAccumulatedFrame()) ?? false {
+                gameEngine?.removeEnemy(enemyEntity)
+            }
+        }
+    }
+    
+    func stopMovementForDuration(for entity: Entity, duration: TimeInterval) {
+        guard let entityMoveComponent = entity.component(ofType: MoveComponent.self) else {
+            return
+        }
+        
+        let temp = entityMoveComponent.maxSpeed
+        entityMoveComponent.maxSpeed = 0
+        Timer.scheduledTimer(withTimeInterval: duration, repeats: false, block: { _ in
+            entityMoveComponent.maxSpeed = temp
+        })
+        
+        guard let entitySpriteComponent = entity.component(ofType: SpriteComponent.self),
```

Comment:
Better way to do this will be to let the `spriteSystem` handle the guard checks, for the `animationNodeKey` we should use a constant to define the movement animation

Reply:
What do u mean by this uh?